### PR TITLE
add in setConfig method to service to allow 'local' config to be set …

### PR DESCRIPTION
**Pull request by chrisweight**
Allows a developer to configure the GF component in the Run phase by calling:
`lnGravityFormService.setConfig(data)`
in the situation where the data is not available in the Config phase.

Please do not delete this branch - component still in development.

(And I'm aware code style is not exactly as-per style guidelines, I'm adjusting this as I go :+1: )

### CODE REVIEW

File: lib/organisms/gravity-form/gravity-form.controller.js [hunk 1]
```diff
@@ -127,21 +127,31 @@ function lnOGravityFormController($log, $scope, $element, $attrs, $sce, lnOGravi
...
+    var unbindInit = $scope.$on('lnOGravityFormConfigUpdated', init);
+  } else {
+    init();
   }
 
+  function init() {
+    $log.debug('lnOGravityFormController -> ', vm, '/ $element: ', $element);
 
-  vm.form.id = 'gform_' + vm.id;
+    unbindInit();
```
Comment (adamf321) [inline lib/organisms/gravity-form/gravity-form.controller.js]: Do we first need to check unbindInit exists? For the case where the app is bootstrapped immediately.

File: lib/organisms/gravity-form/gravity-form.controller.js [hunk 2]
```diff
@@ -127,21 +127,31 @@ function lnOGravityFormController($log, $scope, $element, $attrs, $sce, lnOGravi
...
 
-  $element.on('submit', _submit);
+    if (vm.loadFromAPI === true) {
```
Comment (mitogh) [inline lib/organisms/gravity-form/gravity-form.controller.js]: `vm.loadFromAPI` returns other value rather than `true` or `false` ?

Comment (chrisweight) [inline lib/organisms/gravity-form/gravity-form.controller.js]: No, its should only ever be true or false

File: lib/organisms/gravity-form/gravity-form.controller.js (hunk 1 above)
Comment (chrisweight) [inline lib/organisms/gravity-form/gravity-form.controller.js]: done

File: lib/organisms/gravity-form/gravity-form.controller.js [hunk 3]
```diff
@@ -127,21 +127,33 @@ function lnOGravityFormController($log, $scope, $element, $attrs, $sce, lnOGravi
...
+  function init() {
+    $log.debug('lnOGravityFormController -> ', vm, '/ $element: ', $element);
 
-  vm.form.id = 'gform_' + vm.id;
+    if (angular.isDefined(unbindInit)) {
+      unbindInit();
+    }
 
-  $element.on('submit', _submit);
+    if (vm.loadFromAPI === true) {
```
Comment (mitogh) [inline lib/organisms/gravity-form/gravity-form.controller.js]: Ok based on your last comment I don't think there is a reason to make a strict comparison here with  `if (vm.loadFromAPI) {` should be enough.

**Pull request merged**